data/scripts/dependency-resolver.py

```diff
@@ -257,20 +263,40 @@ def handle_exec_check(args, conf, context):
     path = which(exe)
     required = conf.get("required")
 
-    success = bool(path)
+    if exact_ver or max_ver or atleast_ver:
+        if not cmd:
+            context.info("Could not parse dependency: %s, version requested "
+                "but no version-command to fetch it was specified.", dep)
+            exit(1)
 
-    if required and not success:
-        req_label = context.find_makefile_var("NOT_FOUND")
-        req_label += "executable: %s\\n" % exe
-        context.add_append_makefile_var("NOT_FOUND", req_label, True)
+        result, status = run_command(cmd, context)
+        if not status:
+            ver_match = False
+        elif (exact_ver and result != exact_ver) or \
+            (max_ver and result > max_ver) or \
+            (atleast_ver and result < atleast_ver):
+            version = result
+            ver_match = False
+
+    if required:
+        if not path:
+            req_label = context.find_makefile_var("NOT_FOUND")
+            req_label += "executable: %s\\n" % exe
+            context.add_append_makefile_var("NOT_FOUND", req_label, True)
+        elif not ver_match:
+            req_label = context.find_makefile_var("NOT FOUND")
+            req_label += "%s version: %s\\n" % (exe, version)
```

Comment:
It could, I'm just following the patter used for pkg-config and python checks.
It would need to check what is the kind of check and the expected version. As I said in other comment, it doesn't even log anything for any case we have so far. So I'm not much concerned about it. It's easy to find this information on dependencies file (as easy as to check on this log message).